### PR TITLE
Moves model uuid type from domain/model to core.

### PR DESCRIPTION
**Pull request by tlm**
We have a lot of usage of model uuid as a core type in many packages. It makes more sense for this type to live in core and be referenced from a safer place that doesn't bring in other imports.

This commit is a quick deck chair shuffle.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

This is just a package change. If it compiles it will be fine.

## Documentation changes

N/A

## Links

**Jira card:** JUJU-5341

### CODE REVIEW

**Review by wallyworld — COMMENTED** [review]
I get this is just moving code. But I don't see the reason why we are using a different UUID type just for model.

**Review by wallyworld — APPROVED** [review]
As discussed face to face, we can land this (existing code) but we'll need to start using a common UUID instead of a custom one for model only

File: apiserver/facades/client/cloud/register.go [hunk 1]
```diff
@@ -34,8 +33,8 @@ func newFacadeV7(context facade.Context) (*CloudAPI, error) {
 		WithLegacyUpdater(systemState.CloudCredentialUpdated).
 		WithLegacyRemover(systemState.RemoveModelsCredential)
 
-	credentialCallContextGetter := func(ctx stdcontext.Context, modelUUID model.UUID) (service.CredentialValidationContext, error) {
-		modelState, err := context.StatePool().Get(string(modelUUID))
+	credentialCallContextGetter := func(ctx stdcontext.Context, modelUUID coremodel.UUID) (service.CredentialValidationContext, error) {
```
[the remark this answers is not in the document]
Comment (tlm) [reply apiserver/facades/client/cloud/register.go:36]: I did this on purpose. Wanted it to be explicit to the reader.

**Pull request merged**